### PR TITLE
Bug 1940905: [release-4.6] Remove master as default branch and use only when ref is given

### PR DESCRIPTION
**Pull request by sahil143**
This PR is manual cherry-pick of https://github.com/openshift/console/pull/7730

### CODE REVIEW

**Review by rottencandy — APPROVED** [review]
/lgtm

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
Apart from moving from using `master` as default branch, this PR also introduces support for contextDir in builder image detection but it only does that for github service and not for gitlab or bitbucket services. That fix was added as part of this https://github.com/openshift/console/pull/7470 originally and its a much bigger change including all the tests and test files. The current PR includes part of the changes from https://github.com/openshift/console/pull/7470 and https://github.com/openshift/console/pull/7730. Not sure if all of the changes should be part of this backport. If it's not critical enough we can just push the changes around `master` -> `main` branch issue and keep the PR footprint smaller. Or we include all the changes from both the PRs and backport the whole fix to 4.6. @christianvogt WDYT?

File: frontend/packages/git-service/src/types/git.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export interface GitSource {
   secretType?: SecretType;
   secretContent?: any;
   ref?: string;
+  contextDir?: string;
```
Comment (rohitkrai03) [inline frontend/packages/git-service/src/types/git.ts:14]: Do we need these type changes?

File: frontend/packages/git-service/src/types/repo.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export interface RepoMetadata {
   host: string;
   defaultBranch: string;
   fullName?: string;
+  contextDir?: string;
```
Comment (rohitkrai03) [inline frontend/packages/git-service/src/types/repo.ts:7]: Do we need these type changes?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/git-service/src/services/github-service.ts [hunk 3]
```diff
@@ -69,15 +69,15 @@ export class GithubService extends BaseService {
 
   getRepoFileList = async (): Promise<RepoFileList> => {
     try {
-      const resp = await this.client.git.getTree({
-        // eslint-disable-next-line @typescript-eslint/camelcase
-        tree_sha: this.metadata.defaultBranch,
+      const resp = await this.client.repos.getContents({
         owner: this.metadata.owner,
         repo: this.metadata.repoName,
+        path: '',
```
Comment (rottencandy) [inline frontend/packages/git-service/src/services/github-service.ts:75]: The API doc says: 
>If you omit :path, you will receive the contents of the repository's root directory.

https://octokit.github.io/rest.js/v18#repos-get-contents
Would it be possible to leave out the path?

Comment (rottencandy) [reply frontend/packages/git-service/src/services/github-service.ts:75]: Just checked. There is a lint error on removing, so not possible.

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Review by rottencandy — APPROVED** [review]
/lgtm

**Comment by rottencandy** [comment]
/lgtm cancel

Go builder image is detected for Java sample repository.
asdfasdf

**Comment by sahil143** [comment]
> Go builder image is detected for Java sample repository.

The same issue happens on master as well. May need another issue to track it.

**Pull request merged**